approvestep.php

```diff
@@ -49,18 +51,29 @@
 $PAGE->set_heading(get_string('pluginname', 'lifecyclestep_adminapprove'));
 $PAGE->set_title(get_string('pluginname', 'lifecyclestep_adminapprove'));
 
-if (count($ids) > 0 && ($action == 'proceed' || $action == 'rollback')) {
+if ($action) {
     require_sesskey();
 
-    $sql = 'UPDATE {lifecyclestep_adminapprove} ' .
-            'SET status = ' . ($action == 'proceed' ? 1 : 2) . ' ' .
-            'WHERE id IN (' . implode(',', $ids) . ') ' .
-            'AND status = 0';
-    $DB->execute($sql);
-
+    if (is_array($ids) && count($ids) > 0 && ($action == PROCEED || $action == ROLLBACK)) {
+        $sql = 'UPDATE {lifecyclestep_adminapprove} ' .
+                'SET status = ' . ($action == PROCEED ? 1 : 2) . ' ' .
+                'WHERE id IN (' . implode(',', $ids) . ') ' .
```

Comment:
Be careful with statements like that! They might allow sql injections. For this reason moodle has functions such as $DB->get_in_or_equal(), which transfers an input variable to a sql block and associated parameters.